backend/numpy_backend.py

```diff
@@ -25,16 +26,44 @@ def context(tensor):
         return {'dtype': tensor.dtype}
 
     @staticmethod
-    def tensor(data, dtype=None):
-        return np.array(data, dtype=dtype)
+    def tensor(data, dtype=None, format="dense"):
+        """
+        Parameters
+        ----------
+        data: the input multidimentional array.
+        dtype: datatype
+        format: a string denoting the tensor datatype.
+            if "dense", then return a dense tensor.
+            if "coo", then return a sparse tensor in the COO format.
+        """
+        if format == "dense":
+            return np.array(data, dtype=dtype)
+        elif format == "coo":
+            return sparse.COO.from_numpy(np.array(data, dtype=dtype))
```

Comment:
We may need further abstraction for sparse tensors, but currently it's unclear. We can move forward along the way.

Reply:
agreed